### ISSUE TITLE
proxy_set_header X-Forwarded-Host in advance configuration doesn't take effect

### ISSUE THREAD

**Issue by awsong** [jc21/nginx-proxy-manager#1483]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**

I found that if I put 
`proxy_set_header X-Forwarded-Host $http_host`
in my advance config it doesn't take effect. After looking at the saved config file I found that the statement is put outside of the `location` block. If I manually move the statement into `location` block it works fine.

I also checked Custom Nginx Configurations section in the documentation, it seems that there is no option to let me customize inside `location` block.

![image](https://user-images.githubusercontent.com/787544/137233932-5b0b0464-ef18-4199-b3b6-bbc27b4a5339.png)
![image](https://user-images.githubusercontent.com/787544/137233978-ed85fd00-b3ff-4a3c-9007-df285c4728b0.png)

**Nginx Proxy Manager Version**

2.9.9

**To Reproduce**

**Expected behavior**

**Screenshots**

**Operating System**

**Additional context**

**Comment by chaptergy** [comment]
Yeah, that's how the advanced config works. I think there is some weird behavior in nginx where it doesn't apply any headers from outside the location block if there is even a single header set inside the location block. However you should be able to add a custom location `/` where you can duplicate the stuff you need and add your custom header.

**Comment by Horuyuki** [comment]
> Yes you are right. I can add custom locate / in the settings and it works fine.

Would you mind letting me know how to do it?

I want to access it at / and putting the lines at with that in mind, forwarded to the same ip and the parent proxy host does not solve the problem for me. Can I see what you did?

**Comment by awsong** [comment]
> > Yes you are right. I can add custom locate / in the settings and it works fine.
> 
> Would you mind letting me know how to do it?
> 
> I want to access it at / and putting the lines at with that in mind, forwarded to the same ip and the parent proxy host does not solve the problem for me. Can I see what you did?

Here is how I did it:
![image](https://user-images.githubusercontent.com/787544/140264958-766d2ae3-3243-4663-8edc-3a5541e06bb2.png)

**Comment by Horuyuki** [comment]
> 
> 
> > > Yes you are right. I can add custom locate / in the settings and it works fine.
> > 
> > 
> > Would you mind letting me know how to do it?
> > I want to access it at / and putting the lines at with that in mind, forwarded to the same ip and the parent proxy host does not solve the problem for me. Can I see what you did?
> 
> Here is how I did it: ![image](https://user-images.githubusercontent.com/787544/140264958-766d2ae3-3243-4663-8edc-3a5541e06bb2.png)

Thank you, but apparently, my apps worked without needing to insert these but only enabling websocket. Not a professional so don't know the difference except a few lines being added. But if I do need some custom lines I will be sure to add them that way in the future.

Thank you so much!
